packages/share/ios/Classes/SharePlugin.m

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "SharePlugin.h"
+
+static NSString *const PLATFORM_CHANNEL = @"plugins.flutter.io/share";
+
+@implementation SharePlugin
+
+- (instancetype)initWithController:
+    (FlutterViewController *)controller {
+  FlutterMethodChannel *shareChannel = [FlutterMethodChannel
+      methodChannelWithName:PLATFORM_CHANNEL
+            binaryMessenger:controller];
+
+  [shareChannel setMethodCallHandler:^(FlutterMethodCall *call,
+                                       FlutterResult result) {
+    if ([@"share" isEqualToString:call.method]) {
+      UIActivityViewController *activityViewController =
```

Comment:
Similary here as for SharePlugin.java, you could consider moving out the platform-specific code.